src/NServiceBus.NHibernate/SynchronizedStorage/NHibernateLazyNativeTransactionSynchronizedStorageSession.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace NServiceBus.Persistence.NHibernate
+{
+    using System;
+    using System.Threading.Tasks;
+    using global::NHibernate;
+    using Janitor;
+
+    [SkipWeaving]
+    class NHibernateLazyNativeTransactionSynchronizedStorageSession : CompletableSynchronizedStorageSession, INHibernateSynchronizedStorageSession
+    {
+        Func<ISession> sessionFactory;
+        ISession session;
+        ITransaction transaction;
+
+        public NHibernateLazyNativeTransactionSynchronizedStorageSession(Func<ISession> sessionFactory)
+        {
+            this.sessionFactory = sessionFactory;
+        }
+
+        public ISession Session
+        {
+            get
+            {
+                if (session == null)
```

Comment:
I personally prefer this for readability:

```
public ISession Session
{
    get { return lazySession.Value; }
}
```

We don't need the thread safety indeed but that overhead can be prevented by using the correct overload.